internal/terraform/node_resource_plan_instance.go

```diff
@@ -155,7 +157,8 @@ func (n *NodePlannableResourceInstance) managedResourceExecute(ctx EvalContext)
 		}
 	}
 
-	importing := n.importTarget.ID != ""
+	// If the resource is already in state, do not attempt to import it again.
+	importing := n.importTarget.ID != "" && resourceInPriorState == nil
```

Comment:
I think this check is fine, but having this still listed in the imports is causing other side effects like:
- planning with missing config will panic because we got here with the intent of importing but there is state
- the plan will still show `(imported from "id string")`
- replacing the instance will show `Warning: this will destroy the imported resource`

We should prevent adding this to the import process at all when it already exists in the state.